### PR TITLE
GraphNG: Fix thresholds by  color not following data update

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

This PR tries to fix https://github.com/grafana/support-escalations/issues/2441 by passing down allFrames and a dynamicSeriesColor function to the SeriesBuilder this way when an update happens to the data it can react to color changes as well. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/support-escalations/issues/2441 

**Special notes for your reviewer**:

<details>
<summary>Dashboard for repro</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 277,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "line",
            "fillOpacity": 0,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "none"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "#EAB839",
                "value": 20
              },
              {
                "color": "red",
                "value": 50
              },
              {
                "color": "#6ED0E0",
                "value": 70
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom"
        },
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "targets": [
        {
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "random_walk"
        },
        {
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "B",
          "scenarioId": "random_walk"
        }
      ],
      "title": "Panel Title",
      "type": "timeseries"
    }
  ],
  "schemaVersion": 36,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "Threshold",
  "uid": "4uVKcUwnk",
  "version": 2,
  "weekStart": ""
}
```
</details>

### CODE REVIEW

File: packages/grafana-ui/src/components/GraphNG/utils.test.ts [hunk 1]
```diff
@@ -133,6 +146,9 @@ function mockDataFrame() {
   const f5Config: FieldConfig<GraphFieldConfig> = {
     displayName: 'Metric 4',
     decimals: 2,
+    color: {
```
Comment (zoltanbedi) [inline packages/grafana-ui/src/components/GraphNG/utils.test.ts:149]: Had to change and add this as `getFieldColorModeForField` returns threshold as default if color.mode is not set.

**Comment by leeoniya** [comment]
hey @zoltanbedi

pushed a couple bug fixes and simplified somewhat, please take a look!

**Review by leeoniya — APPROVED** [review]
i think this is okay after my fixes, but please double-check i haven't re-broken your fixes and that i havent broken something new in `graph-ng` panels that demo various coloring, fill modes & thresholds.

image

**Comment by zoltanbedi** [comment]
Thanks @leeoniya, still works as expected!

**Pull request merged**